### ISSUE TITLE
EXC_BAD_ACCESS with errorBlock

### ISSUE THREAD

**Issue by guyschlider** [clayallsopp/ShotBlocker#6]
Hi,

I guess when the user denies the photos permission the errorBlock should be called, but I can see on crashlytics crashes that says
EXC_BAD_ACCESS KERN_INVALID_ADDRESS at 0x0000000000000010
coming from this part of the code:

failureBlock: ^(NSError *error) {
    if (self.errorBlock) {
      dispatch_async(dispatch_get_main_queue(), ^() {
        self.errorBlock(error);
      });
    }

Any ideas?

**Comment by clayallsopp** [comment]
Yeah - at a glance there are some problems with my original implementation. One is that `self` is referenced inside that block without being `weak`-ified, i.e.

```
__weak ShotBlocker *weakSelf = self;
... failureBlock: ^(NSError *error) {
  if (weakSelf.errorBlock) {
    // sort of thing
  }
}];
```

(or maybe keep the failureBlock around as a strong-referenced ivar?)

There's probably more of these sort of errors in the code. I don't have any time to audit them, but I'd welcome any PRs which addressed them!

**Comment by guyschlider** [comment]
Yeah I actually came up with the same solution after digging into it.

Will try to submit a PR tomorrow. 

thanks alot!
